source/state_representation/src/State.cpp

```diff
@@ -33,12 +33,14 @@ void State::set_filled() {
   this->reset_timestamp();
 }
 
-const std::chrono::time_point<std::chrono::steady_clock>& State::get_timestamp() const {
-  return this->timestamp_;
+double State::get_age() const {
+  return
+      std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::steady_clock::now() - this->timestamp_).count()
```

Comment:
you need to static_cast<double> the count before the division. Otherwise it will just floor it to the nearest second.